packages/langium/src/grammar/grammar-util.ts

```diff
@@ -409,20 +413,51 @@ function resolveTransitiveImportsInternal(documents: LangiumDocuments, grammar:
     return Array.from(grammars);
 }
 
+export function createServicesForGrammar(grammar: string, module?: Module<LangiumServices>, sharedModule?: Module<LangiumSharedServices>): LangiumServices {
```

Comment:
Proposal: wrap `module` and `sharedModule` into a `config` object and add more optional properties to make these customizable / reusable: `grammarServices`, `parserConfig`, `languageMetaData`. When not provided, fall back to the defaults you specified here.

Reply:
Great idea, that allows us to easily reuse this function in the `langium-cli` for parser validation 👍